object_store/src/dummy.rs

```
@@ -0,0 +1,121 @@
+//! Crate that mimics the interface of the the various object stores
```

I think we should fail when creating the store, not when using it. I mean the server probes the store anyways on startup (due to the issue w/ the late istio startups) and will bail if it cannot use the store. But from a control-flow perspective I would find that way easier. That would also simplify the code here a bit since you can keep `ObjectStore::new_amazon_s3` and multiplex there to either use the real code or return an error (I think you can use feature-ifs on code-blocks).

---

I believe this PR implements the behavior you prefer (fail fast if you try and use an object store implementation that didn't have compiled in support):

The only thing that ever tries to use a created `DummyObjectstore` is a test -- in all other cases, trying to instantiate an object store if those crate features are not enabled, results in an error to start such as the following. 

```
alamb@ip-10-0-0-124 influxdb_iox2 % cargo run -- run --object-store=Azure --azure-storage-account="foo" --azure-storage-access-key="bar" --bucket="baz"
cargo run -- run --object-store=Azure --azure-storage-account="foo" --azure-storage-access-key="bar" --bucket="baz"
    Finished dev [unoptimized + debuginfo] target(s) in 0.37s
     Running `target/debug/influxdb_iox run --object-store=Azure --azure-storage-account=foo --azure-storage-access-key=bar --bucket=baz`
Server command failed: Run: Microsoft Azure configuration was invalid: Unconfigured Object Store error: object_store crate of type 'azure' is not configured
```